libgraph/include/katana/GraphTopology.h

```diff
@@ -598,6 +598,9 @@ class KATANA_EXPORT CondensedTypeIDMap : public GraphTopologyTypes {
   void invalidate() noexcept { is_valid_ = false; };
 
   const EntityTypeID* index_to_type_map_data() const noexcept {
+    if (index_to_type_map_.empty()) {
+      return nullptr;
```

Comment:
Shouldn't this be an error as if we always have at least one entity type ID?

Reply:
I think if you generate a completely empty graph it will have no type ids until the first commit.

Reply:
or at least that was my understanding from looking at the metadata in the RDG

Reply:
Will this be called on empty graphs only in our unit tests or is that a user-visible use-case?

Reply:
I think both directions of this mapping should always have the mapping between index 0 and kUnknownEntityType, like EntityTypeManager always has kUnknownEntityType even for empty graphs. 

Looks like the issue is that we only are going over in use edge EntityTypes, https://github.com/KatanaGraph/katana/blob/f83d4ddf9b5fb925fff54a969d0c4e030211c022/libgraph/src/GraphTopology.cpp#L536
so on an empty graph, kUnknownEntityType won't get included.

Reply:
I don't know if we ever want users to create/use empty graphs, but I think it would be useful if we did explicitly support it. I think this solution isn't the best, and ideally this function should return an optional or something, but right now I just want to unblock our use cases of empty graphs.

Reply:
I'm not sure. I could also see the argument that the CondensedTypeIDMap should be empty on an empty Graph. 
Since the CondensedTypeID map is supposed to only contain of the in-use EntityTypeIDs. 

The more I look at it, the more I think it makes sense for the CondensedTypeIDMap (and so the id_typ_type_map) to be empty for an empty graph, no EntityTypeIDs are in use here, so there is nothing to hold on to.

Reply:
I think we need to decide the semantics. Empty graph issue aside, if graph has nodes without assigning a type to any of the nodes (which is allowed in our system right?), then what is the current and new behavior?

Reply:
no change. There's no functional change in this PR at all, it just make sanitizer stop complaining.

Reply:
Thanks for your patience and clarification guys!

I'm concerned that if we return `nullptr`, it'll lead to crashes in the uses of this function for empty graphs because the users might NOT check whether it's empty or not. 

@mindv0rtex Can you please add a TODO/doxygen-comments to clarify this in a subsequent PR?

For now, I'll approve to merge this.

Reply:
> I'm concerned that if we return nullptr, it'll lead to crashes in the uses of this function for empty graphs because the users might NOT check whether it's empty or not.

agreed, we should not be returning nullptrs here. Getting pointers to potentially empty vectors leaves us in a bad spot .